src/libsystemd/sd-device/device-monitor.c

```diff
@@ -450,6 +459,25 @@ static int passes_filter(sd_device_monitor *m, sd_device *device) {
         return device_match_parent(device, m->match_parent_filter, m->nomatch_parent_filter);
 }
 
+static bool check_sender_uid(sd_device_monitor *m, uid_t uid, bool is_unicast) {
+        assert(m);
+
+        /* Always trust messages from uid 0. */
+        if (uid == 0)
+                return true;
+
+        /* Trust messages come from outside of the current user namespace. */
+        if (!uid_range_contains(m->uid_range, m->n_uid_range, uid))
+                return true;
+
+        /* Trust unicast messages sent by the same UID we are running. */
+        if (is_unicast && (uid == getuid() || uid == geteuid()))
```

Comment:
hmm, what's the rationale for disallowing this for broadcast msgs?

Reply:
IIUC, broadcast message can be sent by uid 0.

Reply:
sure, that might be implemented like that in kernel, but I see no reason to check for that. I mean, maybe kernel policy changes one day. If it did I see no reason why multicast msgs from same user would *not* be OK.

Reply:
I guess that will never happen. But OK. The condition `is_unicast` is dropped now.